File: C/prime.c
#include <stdio.h>

int main()
{
    int num;
    printf ("Type a number to see if it is prime: ");
    scanf ("%d",&num);

    if (num <= 2){
        printf("%d Isn't a prime number\n", num);
    }
    else{
        int i;
        for (i=2; i<=num/2; i++){
            if (num % i == 0){
                printf("%d isn't a prime number\n", num);
                return(0);
            }
        }
        printf("%d is a prime number\n", num);
        return(0);
    }
}

File: Haskell/prime.hs
prime :: Int -> Bool
prime n
  | n < 2     = False
  | n == 2    = True
  | otherwise = null [x | x <- [2..(n-1)], n `mod` x == 0]

main :: IO ()
main = do
  putStrLn "Type a number:"
  input <- getLine
  let num = read input :: Int
  if prime num
    then putStrLn $ show num ++ " is prime."
    else putStrLn $ show num ++ " isn't prime."